cloud/services/publicips/service.go

```diff
@@ -16,20 +16,30 @@ limitations under the License.
 
 package publicips
 
-import (
-	"sigs.k8s.io/cluster-api-provider-azure/cloud/scope"
-)
+import azure "sigs.k8s.io/cluster-api-provider-azure/cloud"
 
-// Service provides operations on azure resources
+// PublicIPScope defines the scope interface for a public IP service.
+type PublicIPScope interface {
+	azure.ClusterInformation
+	PublicIPSpec() []Spec
+}
+
+// Spec defines the specification for a public IP.
+type Spec struct {
+	Name    string
+	DNSName string
+}
+
+// Service provides operations on Azure resources.
 type Service struct {
-	Scope *scope.ClusterScope
+	Scope PublicIPScope
 	Client
 }
 
 // NewService creates a new service.
-func NewService(scope *scope.ClusterScope) *Service {
+func NewService(scope PublicIPScope) *Service {
 	return &Service{
 		Scope:  scope,
```

Comment:
There are too many things called `Scope`, I'm getting confused keeping track which is which